devel/python/python/ert/enkf/local_dataset.py

```diff
@@ -0,0 +1,57 @@
+from ert.cwrap import BaseCClass, CWrapper
+from ert.enkf import ENKF_LIB
+from ert.ecl import EclRegion
+
+class LocalDataset(BaseCClass):
+
+    def __init__(self, name):
+        raise NotImplementedError("Class can not be instantiated directly!")
+                 
+    def getName(self):
+        return LocalDataset.cNamespace().name(self)
+    
+    def getActiveList(self, key):
+        return LocalDataset.cNamespace().active_list(self, key)
+
+    def addNodeWithIndex(self, key, index):
+        assert isinstance(key, str)
+        assert isinstance(index, int)
+        
+        LocalDataset.cNamespace().add_node(self, key)              
+        active_list = self.getActiveList(key)
+        active_list.addActiveIndex(index)
+                
+    def addField(self, key, ecl_region):
+        assert isinstance(key, str)
+        assert isinstance(ecl_region, EclRegion)
+        
+        LocalDataset.cNamespace().add_node(self, key)
+        active_list = self.getActiveList(key)
+        
+        active_region = ecl_region.getActiveList()
+        
+        for i in active_region:
+            active_list.addActiveIndex(i)
+                            
+    def free(self):
+        LocalDataset.cNamespace().free(self)
+
+
+
+cwrapper = CWrapper(ENKF_LIB)
+cwrapper.registerType("local_dataset", LocalDataset)
```

Comment:
Use the method `registerObjectType` which is a convenience wrapper registering all three types in one go.